core/src/main/java/org/apache/iceberg/BaseUpdatePartitionSpec.java

```diff
@@ -242,7 +242,7 @@ public PartitionSpec apply() {
         // field IDs were not required for v1 and were assigned sequentially in each partition spec starting at 1,000.
         // to maintain consistent field ids across partition specs in v1 tables, any partition field that is removed
         // must be replaced with a null transform. null values are always allowed in partition data.
-        builder.add(field.sourceId(), field.fieldId(), field.name(), Transforms.alwaysNull());
+        builder.add(field.sourceId(), field.fieldId(), field.name() + "_" + field.fieldId(), Transforms.alwaysNull());
```

Comment:
Why do this here? I think it is only necessary if the field is later replaced.

Reply:
I really just wanted to keep it consistent, just make all null transforms have this little extra bit of naming. I was hoping I could make that regardless of the operations that were performed the name of the transform would end up being the same.

Reply:
I think that this is consistent for us, but causes unexpected changes for users because columns they don't expect to change will be renamed.